src/App.tsx

```diff
@@ -1,39 +1,55 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    document.addEventListener('contextmenu', (clickHandle) => {
+      clickHandle.preventDefault(); // not to show the context menu
+      this.setState({ hasClock: false });
```

Comment:
Assign a callback to a variable so that later in componentWillUnmount() remove the handling of this event.
Example:
    document.addEventListener('contextmenu', this.VARIABLE);